### PR TITLE
Suspend and resume instead of restart video stream

### LINKED ISSUE #979: Suspend and resume instead of restart video stream
### Issue Report
Currently the streaming media manager closes and reopens the video service and recreates a new video encoder in case the app moves from Background to Active (visible) application state on the phone. This issue improves the video stream by keeping the video stream open recreating the video encoder only.

##### Reproduction Steps
1. Start the MobileNav app on the iPhone
2. Connect the phone to an SDL head unit and select the app after it registers
3. Wait for the app to start streaming video to the head unit
4. Press Home button on the phone
5. Reopen the app on the phone

##### Expected Behavior
At 4. The SDL media manager should only suspend the video stream
At 5. The app should resume the video stream immediately

##### Observed Behavior
Because the SDL media manager restarts the video stream which takes a few seconds

##### OS & Version Information
* SDL iOS Version: 5.2

### PR DESCRIPTION
**Pull request by ghost**
Fixes #979 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes:
- New video stream state Suspended added. It'll occur if the stream was ready and the app entered inactive state
- Property `shouldRestartVideoStream` is deprecated as not used anymore and can be deleted with the next major version.

### Testing Plan
There was a unit test if the app enters inactive state. The test was changed to match the new state.

##### Enhancements
* Resuming a suspended video stream is faster (<1s) than restarting it (>1s)

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLStreamingMediaLifecycleManager.m [hunk 1]
```diff
@@ -434,6 +442,10 @@ - (void)didEnterStateVideoStreamReady {
     }
 }
 
+- (void)didEnterStateVideoStreamSuspended {
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLStreamingMediaLifecycleManager.m]: This method is unused

**Comment by NicoleYarroch** [comment]
Every time the video comes to the foreground, a `SDLVideoStreamDidStartNotification` is broadcast. However, no corresponding `SDLVideoStreamDidStopNotification` is sent. Perhaps there should be another notification called `SDLVideoStreamSuspendedNotification` that is broadcast when the video stream is suspended.

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
I left one comment on the PR about fixing notifications.

File: SmartDeviceLink/SDLStreamingMediaLifecycleManager.m (hunk 1 above)
Comment (kshala-ford) [reply SmartDeviceLink/SDLStreamingMediaLifecycleManager.m]: True. I wasn‘t sure if the state methods *must* be implemented. I can remove it if it‘s not necessary.

Comment (kshala-ford) [reply SmartDeviceLink/SDLStreamingMediaLifecycleManager.m]: @NicoleYarroch I added the requested suspended notification to the method.

**Review by NicoleYarroch — APPROVED** [review]

**Pull request merged**